File: leetcode2025/2025_01/leetcode_2239_findClosestNumber.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int res = 0, dis = INT_MAX;
        for (int i : nums){
            if (abs(i) < dis){
                dis = abs(i);
                res = i;
            }
            else if (abs(i) == dis){
                res = max(res, i);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {-4,-2,1,4,8};
    cout << s.findClosestNumber(nums) << endl;
}

File: 04.Simulation模拟/leetcode_2239_findClosestNumber.py
from typing import List

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        res, dis = 0, float("inf")
        for i in nums:
            if abs(i) < dis:
                dis = abs(i)
                res = i
            elif abs(i) == dis:
                res = max(res, i)
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [-4,-2,1,4,8]
    print(s.findClosestNumber(nums))
